ui-v2/src/components/ui/state-badge/index.tsx

```diff
@@ -0,0 +1,68 @@
+import { cn } from "@/lib/utils";
+import { Badge } from "../badge";
+import type { components } from "@/api/prefect";
+import {
+	ClockIcon,
+	PauseIcon,
+	XIcon,
+	CheckIcon,
+	ServerCrashIcon,
+	BanIcon,
+	PlayIcon,
+} from "lucide-react";
+
+const ICONS = {
+	COMPLETED: CheckIcon,
+	FAILED: XIcon,
+	RUNNING: PlayIcon,
+	CANCELLED: BanIcon,
+	CANCELLING: BanIcon,
+	CRASHED: ServerCrashIcon,
+	PAUSED: PauseIcon,
+	PENDING: ClockIcon,
+	SCHEDULED: ClockIcon,
+} as const;
```

Comment:
> The states are accessible in StateType in the api spec. Can we use that with some typescript magic so that we get alerted when new states get thrown about?

This should do the trick: 

```suggestion
} as const satisfies Record<components["schemas"]["StateType"], unknown>;
```